rotkehlchen/interfaces.py

```diff
@@ -70,20 +72,73 @@ def can_query_history(
         """Checks if it's okay to query historical price"""
 
     @abc.abstractmethod
-    def query_historical_price(
+    def _query_historical_price(
             self,
             from_asset: Asset,
             to_asset: Asset,
             timestamp: Timestamp,
     ) -> Price:
-        """An oracle implements this to return a historical price from_asset to to_asset at time.
+        """Implementation specific historical price query logic.
 
         If no price can be found may raise:
         - PriceQueryUnsupportedAsset
         - NoPriceForGivenTimestamp
         - RemoteError
         """
 
+    @abc.abstractmethod
+    def oracle_source(self) -> HistoricalPriceOracle:
```

Comment:
So for these new ones can't you turn them into a constant?

```python
class Base(ABC):
    @abstractmethod
    def CONSTANT(self) -> Final[int]: ...

class Child(Base):
    CONSTANT: Final[int] = 42
```

Essentially you would need to add the `Final` typing and initialize at all Children oracle class definitions.
This is more performant as it's stored directly in the class's namespace rather than being computed via property/method calls.